internal/rbd/nodeserver.go

```
@@ -291,14 +273,14 @@ func (ns *NodeServer) NodeStageVolume(
 	}
 	defer ns.VolumeLocks.Release(volID)
 
-	// Check this is a migration request because in that case, unlike other node stage requests
-	// it will be missing the clusterID, so fill it by fetching it from config file using mon.
-	if req.GetVolumeContext()[intreeMigrationKey] == intreeMigrationLabel && req.VolumeContext[util.ClusterIDKey] == "" {
-		cID, cErr := getClusterIDFromMigrationVolume(req.GetVolumeContext()["monitors"])
-		if cErr != nil {
-			return nil, status.Error(codes.Internal, cErr.Error())
+	// Check this is a migration request and verify clusterID in the volumeContext.
+	if isMigrationContext(req.GetVolumeContext()) {
+		// verify and parse volID.
+		if req.GetVolumeContext()[util.ClusterIDKey] == "" {
+			log.ErrorLog(ctx, "clusterID is nil in the volume context")
+
+			return nil, status.Error(codes.InvalidArgument, "missing required parameter clusterID")
```

IMO this check can be removed as the clusterID validation will be done when populating the rbdVolume just like other NodeStageRequest, we don't need to tread this as special

---

Thought an extra check here is good for fail early, but removing it.